src/cryptoadvance/specter/controller.py

```diff
@@ -467,21 +468,122 @@ def new_wallet(wallet_type):
+                        address_type = 'sh-wsh'
+                    elif descriptor.sh:
+                        address_type = 'sh-wsh'
+                    else:
+                        address_type = 'pkh'
+                    keys = []
+                    cosigners = []
+                    unknown_cosigners = []
+                    if sigs_total == None:
+                        sigs_total = 1
+                        sigs_required = 1
+                        descriptor.origin_fingerprint = [descriptor.origin_fingerprint]
+                        descriptor.origin_path = [descriptor.origin_path]
+                        descriptor.base_key = [descriptor.base_key]
+                    for i in range(sigs_total):
+                        cosigner_found = False
+                        for device in app.specter.device_manager.devices:
+                            cosigner = app.specter.device_manager.devices[device]
+                            for key in cosigner.keys:
+                                if key.fingerprint + key.derivation.replace('m', '') == \
+                                    descriptor.origin_fingerprint[i] + descriptor.origin_path[i].replace("'", 'h'):
+                                    keys.append(key)
+                                    cosigners.append(cosigner)
+                                    cosigner_found = True
+                                    break
+                            if cosigner_found:
+                                break
+                        if not cosigner_found:
+                            desc_key = Key.parse_xpub('[{}{}]{}'.format(
+                                descriptor.origin_fingerprint[i],
+                                descriptor.origin_path[i],
+                                descriptor.base_key[i],
+                            ))
+                            unknown_cosigners.append(desc_key)
+                        #     raise Exception('Could not find device with matching key to import wallet')
+                    wallet_type = 'multisig' if sigs_total > 1 else 'simple'
+                    createwallet = 'createwallet' in request.form
+                    if createwallet:
+                        wallet_name = request.form['wallet_name']
+                        for i, unknown_cosigner in enumerate(unknown_cosigners):
+                            unknown_cosigner_name = request.form['unknown_cosigner_{}_name'.format(i)]
+                            device = app.specter.device_manager.add_device(name=unknown_cosigner_name, device_type='other', keys=[unknown_cosigner])
+                            keys.append(unknown_cosigner)
+                            cosigners.append(device)
+                        wallet = app.specter.wallet_manager.create_wallet(wallet_name, sigs_required, address_type, keys, cosigners)
+                        try:
+                            wallet.cli.rescanblockchain(startblock, timeout=1)
```

Comment:
will fail if the node is pruned - maybe `flash()` an error if startblock < pruneheight and rescan from pruneheight